### ISSUE TITLE
Logging needs to understand dig.Out

### ISSUE THREAD

**Issue by breerly** [uber-go/fx#540]
When Fx's output encounters a `dig.Out`, it should print a log line for *each* dependency being provided in the `dig.Out`.

So instead of:

```
[Fx] PROVIDE        zapfx.Result <= vendor/code.uber.internal/go/zapfx.git.module()
```
It should contain:

```
[Fx] PROVIDE        zap.AtomicLevel <= vendor/code.uber.internal/go/zapfx.git.module()
[Fx] PROVIDE        *zap.Logger <= vendor/code.uber.internal/go/zapfx.git.module()
[Fx] PROVIDE        *zap.SugaredLogger <= vendor/code.uber.internal/go/zapfx.git.module()
```

**Comment by glibsm** [comment]
Definitely. Shouldn't be hard to do, there are examples of that in the dig repo, such as `traverseOutTypes`

Do we want to explode one per line, or keep things combined:

```
[Fx] PROVIDE        zap.AtomicLevel, *zap.Logger, *zap.SugaredLogger <= vendor/code.uber.internal/go/zapfx.git.module()
```

**Comment by breerly** [comment]
Exploded - log lines should be line-oriented, and I want people to be able to easily scan down the column to explore what types are available.

**Comment by glibsm** [comment]
In order to properly implement this `fxlog` packages needs to know what is and what isn't In/Out objects. It's fairly trivial to duplicate that logic, but I don't think that's the best approach in the long term.

I think the right thing to do is to export the currently private `isInObject` and `isOutObject` dig functions: https://github.com/uber-go/dig/blob/master/types.go#L69

**Comment by breerly** [comment]
Suggest just calling it `IsOut` and `IsIn`

**Comment by breerly** [comment]
Resolved in #126
